src/sema/builtin.rs

```diff
@@ -1616,4 +1616,30 @@ impl Namespace {
             Symbol::Function(vec![(pt::Loc::Builtin, func_no)])
         ));
     }
+
+    pub fn add_substrate_builtins(&mut self) {
+        let file_no = self.files.len();
+        self.files.push(File {
+            path: PathBuf::from("substrate"),
+            line_starts: Vec::new(),
+            cache_no: None,
+        });
+
+        // The Hash type from ink primitives.
+        let type_no = self.user_types.len();
+        self.user_types.push(UserTypeDecl {
+            tags: vec![],
```

Comment:
It might be useful to add some documentation tags here. I think what you need is:
```rust
tags: vec![Tag { Tag: "notice".into(), no: file_no, Value: "ink-like hash value" }],
```
This would be used by the language server.